x/nftmarket/keeper/nft_listing.go

```diff
@@ -58,6 +59,53 @@ func getTimeKey(prefix string, timestamp time.Time) []byte {
 	return bz
 }
 
+func (k Keeper) SaveNftListing(ctx sdk.Context, listing types.NftListing) {
+	k.SetNftListing(ctx, listing)
+	k.UpdateClass(ctx, listing)
+}
+
+func (k Keeper) UpdateClass(ctx sdk.Context, listing types.NftListing) {
+	switch listing.State {
+	case types.ListingState_LISTING:
+		k.SetClass(ctx, listing)
+	case types.ListingState_SUCCESSFUL_BID:
+		// todo delete nftid from class
+	}
+}
+
+func (k Keeper) SetClass(ctx sdk.Context, listing types.NftListing) {
```

Comment:
Can you add comment on what SetClass does?

Reply:
I renamed SetClass to SetListedClass.
Did you need comment?

Reply:
I think it's okay now